bundle-test.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -e
+
+yell() { echo "$0: $*" >&2; }
+die() { yell "$*"; exit 111; }
+try() { "$@" || die "cannot $*"; }
+
+function check-error-logs() {
+  #statements
+  printf "CHECKING "$1" for errors ... \n"
+  if [[ -s "$1" ]]; then
+    printf "You must fix "$1" errors before submitting a pull request \n"
+    cat $1
+    exit -1
+  fi
+  printf "No errors in "$1" \n \n"
+}
+
+# Check versions
+cppcheck --version
+
+# Setup standard permissive build
+cp ./cfe/cmake/Makefile.sample Makefile
+cp -r ./cfe/cmake/sample_defs sample_defs -v
+
+# Static code analysis
+#   Below is intent but fails - exit code bug in cppcheck 1.82
+#   - cppcheck --force --inline-suppr --quiet --error-exitcode=1 .
+#   Work around script:
+cppcheck --force --inline-suppr --quiet . 2> cppcheck_err.txt
+
+check-error-logs cppcheck_err.txt
+# if [[ -s cppcheck_err.txt ]]; then
+#   echo "You must fix cppcheck errors before submitting a pull request"
+#   echo "\ "
+#   cat cppcheck_err.txt
+#   exit -1
+# fi
+
+# Prep and build
+make prep
+make
+make install
+
+# Run unit tests and generate coverage results
+make test
+make lcov
+# Eventually check/enforce minimum coverage
+
+# Make documentation
+make doc > make_doc_stdout.txt 2> make_doc_stderr.txt
+
+check-error-logs make_doc_stderr.txt
+
+# Eventually enforce no doxygen warnings
+make usersguide > make_usersguide_stdout.txt 2> make_usersguide_stderr.txt
+
+check-error-logs make_usersguide_stderr.txt
+
+check-error-logs build/doc/warnings.log
+
+make osalguide > make_osalguide_stdout.txt 2> make_osalguide_stderr.txt,
+
+check-error-logs make_osalguide_stderr.txt
+check-error-logs build/doc/warnings.log
```

Comment:
These three lines go together

Reply:
Do you mean to say I should remove the empty line between them?

Reply:
That would be my preference